doc/source/tune/user-guide.rst

```diff
@@ -218,6 +218,68 @@ During training, Tune will automatically log the below metrics in addition to th
 
 All of these metrics can be seen in the ``Trial.last_result`` dictionary.
 
+.. _tune-reproducible:
+
+Reproducible runs
+-----------------
+Exact reproducibility of machine learning runs is hard to achieve. This
+is even more true in a distributed setting, as more non-determinism is
+introduced. For instance, if two trials finish at the same time, the
+convergence of the search algorithm might be influenced by which trial
+result is processed first. This depends on the searcher - for random search,
+this shouldn't make a difference, but for most other searchers it will.
+
+If you try to achieve some amount of reproducibility, there are two
+places where you'll have to set random seeds:
+
+1. On the driver program, e.g. for the search algorithm. This will ensure
+   that at least the initial configurations suggested by the search
+   algorithms are the same.
+
+2. In the trainable (if required). Neural networks are usually initialized
+   with random numbers. Thus you'll want to make sure to set a seed here
+   so that the initialization is always the same.
```

Comment:
```suggestion
2. In the trainable (if required). Neural networks are usually initialized
   with random numbers, and many classical ML algorithms, like GBDTs, make use of
   randomness. Thus you'll want to make sure to set a seed here
   so that the initialization is always the same.
```